app/models/event_schedule.rb

```diff
@@ -31,6 +31,10 @@ def intersecting_events
     room.event_schedules.where(start_time: start_time, schedule: schedule).where.not(id: id)
   end
 
+  def replacement?
+    event.state == 'confirmed' && (!intersecting_events.canceled.empty? || !intersecting_events.withdrawn.empty?)
```

Comment:
@Ana06 yes, that's right.
I like how it is written though, because it makes it clear that we are talking about events that are cancelled or withdrawn, and I think it will protect the result we want to achieve, in case any magic happens in the DB. What do you think?